### ISSUE TITLE
Resolve accessibility violation for validation icon

### ISSUE THREAD

**Issue by kishanivelupillai-sage** [Sage/carbon#5498]
### Current behaviour

Axe reports accessibility issue for validation icon in Textbox component.

**Issue Description:** Ensures ARIA attributes are allowed for an element's role
**Fix the following:** aria-label attribute cannot be used on a span with no valid role attribute.

#### Axe Report Screenshot
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/77396222/191745499-f0456567-e758-4126-b750-ac5724ac1288.png">

### Expected behaviour

Axe does not report accessibility issues for validation icon.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/validation-icon-accessibility-m4fu34

### Carbon Version

109.1.2

### Design Tokens Version

2.40.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by robinzigmond** [comment]
FE-5382
